File: Kotlin/engineeringenglish.kt
fun main() {
    val s = mutableSetOf<String>()

    while (true) {
        try {
            val i = readLine()!!.split(' ')
            var op = ""

            for (w in i) {
                op += if (!s.contains(w.toLowerCase())) w + " " else ". "
                s.add(w.toLowerCase())
            }
            println(op)
        }
        catch (e: Exception) {
            break
        }
    }
}

File: Python/engineeringenglish.py
s = set()

while True:
    try:
        r = input().split(' ')
        op = ""

        for w in r:
            if w.lower() not in s:
                op += w + " "
                s.add(w.lower())
            else:
                op += ". "

        print(op)

    except:
        break